src/entities/Basket/Basket.schema.ts

```diff
@@ -0,0 +1,35 @@
+import * as mongoose from 'mongoose';
+import Basket from './Basket.interface';
+
+const basketSchema = new mongoose.Schema({
+    createdByUserId: { 
+        type: mongoose.Types.ObjectId, 
+        required: true
+    },
+    targetUserID: {
+        type: mongoose.Types.ObjectId,
+        required: true
+    },
+    booksOffered: {
+        type: [mongoose.Types.ObjectId],
+        required: true
+    },
+    booksRequested: {
+        type: [mongoose.Types.ObjectId],
+        required: true
+    },
```

Comment:
every reference should be created using the following schema: createdBy: {type: mongoose.Types.ObjectId, ref: 'User"} ect. We need ref in order to create connections between collections.